source/features/blocks/api/transformers.ts

```diff
@@ -3,31 +3,41 @@ import {
   BlockInfoFragment,
   BlockOverviewFragment,
 } from '../../../../generated/typings/graphql-schema';
+import { lovelacesToAda } from '../../../lib/unit-converters';
 import { IBlockDetailed, IBlockInfo, IBlockOverview } from '../types';
 
 export const blockInfoTransformer = (b: BlockInfoFragment): IBlockInfo => ({
-  id: b.id ? b.id : '',
-  number: b.number ? b.number : 0,
+  id: b.id,
+  number: b.number || 0,
   size: b.size,
   slotNo: b.slotNo ? b.slotNo : 0,
 });
 
 export const blockOverviewTransformer = (
   b: BlockOverviewFragment
 ): IBlockOverview => {
-  const transactions =
-    b.transactions_aggregate &&
-    b.transactions_aggregate.aggregate &&
-    b.transactions_aggregate.aggregate.count
-      ? b.transactions_aggregate.aggregate.count
-      : 0;
+  const blockCreatorPrefix = b.createdBy.substring(0, 11);
+  const createdBy =
+    blockCreatorPrefix === 'SlotLeader-'
+      ? b.createdBy.substring(12, 18)
+      : b.createdBy;
+  const { transactions_aggregate } = b;
   return {
     ...blockInfoTransformer(b),
     createdAt: b.createdAt,
-    createdBy: 'af2800c', // TODO: missing API data
-    epoch: b.epoch ? b.epoch.number : 0,
-    output: 11189.647356, // TODO: missing API data
-    transactions,
+    createdBy,
+    epoch: (b.epoch && b.epoch.number) || 0,
+    output: lovelacesToAda(
+      (transactions_aggregate.aggregate &&
```

Comment:
This shorthand is pretty hard to read. I'm surprised that syntax even works

Reply:
I've updated the imp to use lodash `get`. I did initially intend to use it, but there was some config to enable deep property selection since we're using a webpack plugin

https://github.com/lodash/lodash-webpack-plugin#feature-sets